File: Arquivos/Ad-Hoc/1103_Alarme Despertador.py
while True:
    h1, m1, h2, m2 = map(int,input().split())
    if h1+h2+m1+m2 == 0:
        break
    h = ((h2*60)+m2)-((h1*60)+m1)
    if h<0:
        h = (24*60)+h
    print(h)
        

File: Arquivos JavaScript/Ad-hoc/1103_Alarme Despertador.js
let input = require("fs").readFileSync("stdin","utf8");
let lines = input.split("\n");

for (let i of lines){
    if (i == "0 0 0 0"){
        break;
    }

    let numeros = (i.split(" ")).map((num) => parseInt(num));
    const inicio = { hora: numeros.shift(), minuto: numeros.shift()}
    const fim = { hora: numeros.shift(), minuto: numeros.shift()}

    let diferenca = ((fim.hora*60)+fim.minuto)-((inicio.hora*60)+inicio.minuto)

    if (diferenca < 0){
        diferenca += 24*60
    }
    else if(isNaN(diferenca) || (diferenca === -0)){
        continue;
    }

    console.log(diferenca);
}